### PR TITLE
Navbar dev

### PR DESCRIPTION
**Pull request by caitlinard**
This merge will add in a (rough) sticky navbar.
Run the tests on the trello card here: https://trello.com/c/KuAAmkd2/45-nav-bar

### CODE REVIEW

File: src/components/navbar/navbar.controller.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export default class navbarController {
+    // No current use for a controller here so leaving blank.
```
Comment (cyrillegin) [inline src/components/navbar/navbar.controller.js]: This comment is unnecessary.

File: src/components/navbar/navbar.controller.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    constructor($scope, $window, $timeout) {
```
Comment (cyrillegin) [inline src/components/navbar/navbar.controller.js]: You're not actually using $window

File: src/components/navbar/navbar.controller.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        this.$scope = $scope;
+        this.$window = $window;
+        this.$timeout = $timeout;
+    }
+    $onInit() {
+        console.log('hi bitch');
```
Comment (cyrillegin) [inline src/components/navbar/navbar.controller.js]: No unnecessary console.logs

File: src/components/navbar/navbar.controller.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        this.$timeout(() => {
+            const sticky = $('#navbar-main')[0].offsetTop;
+            window.onscroll = function () {
+                if (window.pageYOffset > sticky) {
+                    $('#navbar-main').addClass('sticky');
+                    $('#main').css('margin-top', '80px');
+                    console.log('here');
```
Comment (cyrillegin) [inline src/components/navbar/navbar.controller.js]: No unnecessary console.logs

File: src/components/navbar/navbar.controller.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    // No current use for a controller here so leaving blank.
+    constructor($scope, $window, $timeout) {
+        this.$scope = $scope;
+        this.$window = $window;
+        this.$timeout = $timeout;
+    }
+    $onInit() {
+        console.log('hi bitch');
+
+        this.$timeout(() => {
```
Comment (cyrillegin) [inline src/components/navbar/navbar.controller.js]: Although this works, it doesn't feel like the correct way to do what you're attempting. Check here for a few ideas: https://stackoverflow.com/questions/2342132/waiting-for-image-to-load-in-javascript
Personally I think attaching an event listener would be the best and easiest way to go through this but up to you. You can leave it as is too, just something to consider.

File: src/components/navbar/navbar.controller.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+                if (window.pageYOffset > sticky) {
+                    $('#navbar-main').addClass('sticky');
+                    $('#main').css('margin-top', '80px');
+                    console.log('here');
+                } else {
+                    $('#navbar-main').removeClass('sticky');
+                    $('#main').css('margin-top', '0');
+                }
+            };
+        }, 2000);
```
Comment (cyrillegin) [inline src/components/navbar/navbar.controller.js]: In regards to the comment about using $timeout, One issue we could have is what if the image takes longer than 2 seconds?

File: src/components/navbar/navbar.style.scss [hunk 7]
```diff
@@ -0,0 +1,57 @@
+@import '../../constants';
+
```
Comment (cyrillegin) [inline src/components/navbar/navbar.style.scss]: There are a ton of unused classes here, please only have classes that this component uses.

File: src/components/navbar/navbar.style.scss [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    overflow: hidden;
+    width: 100%;
+    a {
+        float: left;
+        display: block;
+        text-align: center;
+        padding: 14px;
+        text-decoration: none;
+        &:hover {
+            background-color: #ddd;
```
Comment (cyrillegin) [inline src/components/navbar/navbar.style.scss]: Colors should be variables defined in _constants.scss

File: src/components/navbar/navbar.template.html [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (cyrillegin) [inline src/components/navbar/navbar.template.html]: No DOCTYPE html. If you got an error from the linter, you can turn off that rules in .eslintrc

File: src/pages/home/home.page.html [hunk 10]
```diff
@@ -1,3 +1,10 @@
+<!DOCTYPE html>
```
Comment (cyrillegin) [inline src/pages/home/home.page.html]: No DOCTYPE

File: src/pages/home/home.page.html [hunk 11]
```diff
@@ -1,3 +1,10 @@
...
+
+<img src="/images/kitten.jpg" alt="kitten!" />
```
Comment (cyrillegin) [inline src/pages/home/home.page.html]: alt text should be more descriptive, maybe.

File: src/components/navbar/navbar.controller.js (hunk 5 above)
Comment (caitlinard) [reply src/components/navbar/navbar.controller.js]: I am leaving as is for now, but will continue to consider a more elegant solution

File: src/components/navbar/navbar.controller.js (hunk 6 above)
Comment (caitlinard) [reply src/components/navbar/navbar.controller.js]: I am leaving as is for now, but will continue to consider a more elegant solution

**Pull request merged**